core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -630,5 +632,29 @@ class CityInfo {
         return !isOriginalCapital && !isCapital() && !isHolyCity()
     }
 
+
+    fun cityBorders(): ArrayList<String>{
+        val dimensionalList: ArrayList<TileInfo> = arrayListOf()
+        val cityPositionList: ArrayList<TileInfo> = arrayListOf()
+        val neighbouringCivilizationList: ArrayList<String> = arrayListOf()
+
+        for (tile in getTiles())
+            if (!dimensionalList.contains(tile))
+                dimensionalList.add(tile)
+
+        for (tiles in dimensionalList)
+            for (tile in tiles.neighbors)
+                if (!dimensionalList.contains(tile))
+                    cityPositionList.add(tile)
+
+        for (tile in cityPositionList)
+            if (!neighbouringCivilizationList.contains(tile.getOwner().toString()))
+                neighbouringCivilizationList.add(tile.getOwner().toString())
+
+
+        return neighbouringCivilizationList
```

Comment:
A. You really need to give more meaningful names to the variables here.

B. Don't use CivInfo.toString() if you want to get the civ name, use .civName

C. Looks like two of the values can be considerably shortened:

```kotlin
val dimensionalList = getTiles()
...
val neighbouringCivilizationList = cityPositionList.asSequence().map{ it.getOwner().civName }.distinct()
```

The middle one can also be shortened, but not in so obvious a manner

Reply:
fixed